deepspeed/pt/deepspeed_zero_optimizer.py

```diff
@@ -1087,10 +1199,160 @@ def free_grad_in_param_list(self, param_list):
         for p in param_list:
             p.grad = None
 
+    def get_partition_fp32_group_index(self, group_index):
+        start = self.first_offset[group_index]
+        end = start + self.partition_size[group_index]
+        return [int(start), int(end)]
+
+    #jie:
+    def step_with_cpuoffload(self, closure=None):

```

Comment:
Does it make sense to reuse most of the code in the step? It seems like the code all the way till Line 1261 is common and the code after line 1305 is common across both steps, so there is only about 40 lines of code that is different between the two step. Does it make sense to just put this part in two methods and differentiate them with 'if cpu_offload' ?

Reply:
Done